nautobot_ssot/integrations/librenms/diffsync/adapters/librenms.py

```diff
@@ -67,41 +75,187 @@ def load_location(self, location: dict):
 
     def load_device(self, device: dict):
         """Load Device objects from LibreNMS into DiffSync models."""
+        # Ensure device is a dictionary
+        if not isinstance(device, dict):
+            self.job.logger.warning(f"Device data is not a dictionary: {type(device)} - {device}")
+            self.failed_import_devices.append(device)
+            return
+
+        # Get the hostname field to use
+        hostname_field = (
```

Comment:
Why is this being done again when it's already done for self.hostname_field in the load() call?